pkg/authorization/authorizer/authorizer.go

```diff
@@ -360,6 +383,48 @@ func (a DefaultAuthorizationAttributes) GetUserInfo() user.Info {
 func (a DefaultAuthorizationAttributes) GetVerb() string {
 	return a.Verb
 }
+
+// nonResourceMatches take the remainer of a URL and attempts to match it against a series of explicitly allowed steps that can end in a wildcard
+func (a DefaultAuthorizationAttributes) nonResourceMatches(rule authorizationapi.PolicyRule) bool {
+	for allowedNonResourcePath := range rule.NonResourceURLs {
+		allowPathSteps := splitPath(allowedNonResourcePath)
+
+		for i, currAllowedPathStep := range allowPathSteps {
+			// if we're allowing *, then the path matches
+			if currAllowedPathStep == "*" {
+				return true
+			}
+
+			// if we have run out of steps in the remaining url to match, then this rule does not match
+			if i >= len(a.URLSegments) {
+				break
+			}
+
+			currNonResourceStep := strings.ToLower(a.URLSegments[i])
+			// if the paths don't match, then this rule doesn't match
+			if strings.ToLower(currAllowedPathStep) != currNonResourceStep {
+				break
+			}
+
+			// if we have matched all the way to this point and we're at the end of both paths, then this rule matched exactly
+			if (len(allowPathSteps) == (i + 1)) && (len(a.URLSegments) == (i + 1)) {
+				return true
+			}
+		}
+	}
+
+	return false
+}
+
+// splitPath returns the segments for a URL path.
+func splitPath(path string) []string {
+	path = strings.Trim(path, "/")
```

Comment:
Should we run through path.Clean() first? Not sure what is done in the http handler request building step